### PR TITLE
KAFKA-6012: Close request metrics only after closing request handlers

### PR DESCRIPTION
**Pull request by rajinisivaram**

### CODE REVIEW

**Comment by rajinisivaram** [comment]
@ijuma Can you review, please? Thank you!

File: core/src/main/scala/kafka/server/KafkaServer.scala [hunk 1]
```diff
@@ -547,6 +547,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
 
         if (metrics != null)
           CoreUtils.swallow(metrics.close())
+        if (socketServer != null)
+          CoreUtils.swallow(socketServer.closeRequestMetrics())
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaServer.scala]: Would it make sense to close this in `KafkaRequestHandlerPool`?

Comment (rajinisivaram) [reply core/src/main/scala/kafka/server/KafkaServer.scala]: @ijuma Thank you for the review, updated.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM. Feel free to merge after the tests pass.

**Comment by rajinisivaram** [comment]
@ijuma Thank you for the review, merging to trunk and 1.0.

**Pull request closed without merging**